### PR TITLE
Change delete by index to delete by ID

### PR DESCRIPTION
**Pull request by s0ngyang**

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
 import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
 import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.TypicalIds.ID_FIRST_EMPLOYEE;
 import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_EMPLOYEE;
```
Comment (LordSaumya) [inline src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java:7]: Can delete the `INDEX_FIRST_EMPLOYEE` if not required.

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 2]
```diff
@@ -29,8 +31,16 @@ public class DeleteCommandTest {
...
     @Test
     public void execute_validIndexUnfilteredList_success() {
-        Employee employeeToDelete = model.getFilteredEmployeeList().get(INDEX_FIRST_EMPLOYEE.getZeroBased());
-        DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST_EMPLOYEE);
+        Employee employeeToDelete = null;
+        List<Employee> lastShownList = model.getFilteredEmployeeList();
+        for (Employee employee : lastShownList) {
+            if (employee.getId().equals(ID_FIRST_EMPLOYEE)) {
+                employeeToDelete = employee;
+                break;
```
Comment (LordSaumya) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:39]: Nice work. Can maybe try using either .indexOf or Streams for simplification, but this should be fine.

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
Comment (LordSaumya) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:1]: Can maybe also add a test to check for deleting the same ID twice

**Comment by LordSaumya** [comment]
The rest looks good to me. Get your test cases running and can then merge.

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
Comment (s0ngyang) [reply src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:1]: Added

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java (hunk 2 above)
Comment (s0ngyang) [reply src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:43]: Marked as low priority

**Comment by inezkok** [comment]
lgtm!

**Pull request merged**